scenes/ui/settings_slider/settings_slider.gd

```diff
@@ -2,11 +2,13 @@ class_name SettingsSlider extends MarginContainer
 
 signal data_changed(toggle: bool, value: float)
 
+@export var toggle_button_ui: bool = true
+
 var _toggle: bool
 var _value: float
 
 @onready var toggle_button: Button = %ToggleButton
-@onready var title_label: Label = %TitleLabel
+@onready var title_label: LabelTyping = %TitleLabel
```

Comment:
I wonder if it's possible to organize the scene tree of `settings_screen` something like this:

![image](https://github.com/user-attachments/assets/338e9e7d-71b1-450c-841c-27c4201f67e2)

And then the `settings_slider` checks if there's a `Label` child and then uses that as the `title_label`
Then the shake settings can get a normal `Label` with `ShakeShaderComponent` already attached, and the typing settings can get a `LabelTyping`

(note: this is me trying to apply React reasoning to Godot :P)

Reply:
Hmm, should be possible. 

Would it be better to then only extract the ShakeShaderComponent ? Something like an export variable on the SettingsSlider that selects a target to apply on the internal TitleLabel.

Reply:
Also sounds like an option. Is the idea to also extract the typing label functionality into a TypingComponent? (but then what if you want to apply two effects to a title @_@)

Reply:
Hmm - I'll leave it as is for now. 👀